index.rst

```diff
@@ -1,29 +1,988 @@
+
+.. _automodapi: https://sphinx-automodapi.readthedocs.io/en/latest/
+
+Do not include the handlers in the internal API documentation.
+They won't generate useful entries, and you should not write docstrings for handler functions.
+(They would be redundant with the FastAPI decorator.)
+
+Also do not include ``cli.py`` if you have one.
+Instead, use sphinx-click_ to generate documentation for the command-line interface.
+
+.. _sphinx-click: https://sphinx-click.readthedocs.io/en/latest/
+
+.. _changelog:
+
+Change log
+^^^^^^^^^^
+
+Any application that has a manual should probably also have a change log.
+The change log is maintained in ``CHANGELOG.md`` at the top level of the repository, in Markdown format.
+It should summarize user-visible changes from the previous release.
+
+Each entry should use the following layout:
+
+.. code-block:: markdown
+
+   ## X.Y.Z (YYYY-MM-DD)
+
+   ### Backward-incompatible changes
+
+   - Some change.
+
+   ### New features
+
+   - Some change.
+
+   ### Bug fixes
+
+   - Some change.
+
+   ### Other changes
+
+   - Some change.
+
+Omit any sections that are not needed.
+There will only be backward-incompatible changes for major version bumps and new features for minor version bumps (see :ref:`releases` for more about versioning).
+
+Unlike the normal convention of one sentence per line, each change log bullet point, no matter how many sentences long, should be a single line.
+This allows the change log to be cut and pasted into the text box for the GitHub release description with no formatting changes.
+
+While a release is still being prepared, the date in the version header should instead be ``(unreleased)``.
+Write new change log entries and update the version number based on semantic versioning as changes are merged to save time and ensure a complete change log when preparing the release.
+
+.. _releases:
+
+Releases
 ========
 
-Describes the design pattern and code structure the author uses for FastAPI web applications. This is only one possible approach, and is not a standard or requirement.
+Default to making a new release of the application after every noticable change, including bug fixes.
+Releases are cheap; follow the release early, release often principle.
+
+Each release should publish a Docker image to the GitHub Container Registry.
+(This is generally done via the GitHub Actions configuration provided by the FastAPI Safir App template.)
+Also publishing the container to the Docker Hub registry is not necessary.
+
+Use `Semantic Versioning`_ versioning.
+I'm strict about this: every backwards-incompatible change bumps the major version and every new feature bumps the minor version.
+This means that my version numbers tend to increase faster than a lot of open source software.
+This is fine.
+
+.. _Semantic Versioning: https://semver.org/
 
-Add content here
-================
+Each release should is marked with a Git tag matching the version number (with no leading ``v``).
```

Comment:
Potentially working linking to [setuptools_scm](https://github.com/pypa/setuptools_scm) because it means that the application version is the Git tag.